src/transaction/settings.ts

```diff
@@ -17,9 +17,9 @@ function setTransactionFlags(
   txJSON: TransactionJSON,
   values: FormattedSettings
 ) {
-  const keys = Object.keys(values)
+  const keys = Object.keys(values).filter((key) => AccountFlagIndices[key] !== undefined)
```

Comment:
This filters the keys to be considered to only the ones that are Account Flags. The other keys are non-flag settings, e.g. `domain`, `emailHash`, etc.